lib/p2p/Pool.ts

```diff
@@ -514,12 +517,20 @@ class Pool extends EventEmitter {
       // the node is known, update its listening addresses
       await this.nodes.updateAddresses(peer.nodePubKey!, addresses, peer.inbound ? undefined : peer.address);
     }
+
+    // Check if peer has previously banned us if so set `bannedBy` to false
+    const node = await this.repository.getNode(peerPubKey);
```

Comment:
We shouldn't need to read from the db again ideally, as we would already have all nodes in the `NodeList` class and can just get the node from there.